### ISSUE TITLE
Use of "ECHO OFF" in npm.cmd (etc) causes unwanted side-effects

### ISSUE THREAD

**Issue by peternann** [npm/npm#19681]
#### I'm opening this issue because:
  - [ ] npm is producing an unwanted side-effect

### supporting information:
 - Windows

#### What's going wrong?
The use of "ECHO OFF" in npm.cmd, and related scripts, causes the side-effect of leaving .BAT/.CMD echo off FOR ANY FOLLOWING COMMANDS in a user script

#### How can the CLI team reproduce the problem?

Write a .BAT file with something vaguely useful, perhaps:
```
CALL npm run build
:: Followed by some command you might want to see echoed:
DIR dist
```

Run the .BAT file.

You never see the 'DIR' command line echoed, even though the intention (and expectation) of the 3-line script above is that you would. (You do see the OUTPUT of the command, but that is not the point... In many scripts you want to see the executed command lines to see what's going on).

The problem is that .bat processing is stupid, and most things happen globally unless you explicitly fork a sub-shell with something like "CMD /C"

#### How can the CLI team fix the problem?
Putting "ECHO ON" at the end of the script is NOT the right fix - It can have the reverse side-effect.

Forcing a sub-shell with CMD /C can be achieved in the script, but isn't very clean and has questionable efficiency.

The simplest fix is to REMOVE any "ECHO ON/OFF" commands, and rely on explicit command-by-command silencing. Since the volume of .CMD code in npm is minimal, this is probably reasonable.

The below has been tested as a workable variant of the current npm.cmd file:
Basically - Remove the `@ECHO OFF`, put an `@` at the beginning of any command line, or before any `( command[s] )` sequence used to group multi-line commands. That's it.

```
:: Created by npm, please don't edit manually.

@SETLOCAL

@SET "NODE_EXE=%~dp0\node.exe"
@IF NOT EXIST "%NODE_EXE%" @(
  SET "NODE_EXE=node"
)

@SET "NPM_CLI_JS=%~dp0\node_modules\npm\bin\npm-cli.js"
@FOR /F "delims=" %%F IN ('CALL "%NODE_EXE%" "%NPM_CLI_JS%" prefix -g') DO @(
  SET "NPM_PREFIX_NPM_CLI_JS=%%F\node_modules\npm\bin\npm-cli.js"
)
@IF EXIST "%NPM_PREFIX_NPM_CLI_JS%" @(
  SET "NPM_CLI_JS=%NPM_PREFIX_NPM_CLI_JS%"
)

@"%NODE_EXE%" "%NPM_CLI_JS%" %*
```

While you are there, is the comment on line 1 obsolete now?
It's no longer 'created', it is 'static', no?

**Comment by hcapp01** [comment]
I agree. very annoying bug
